agent/handlers/v1_handlers.go

```diff
@@ -92,6 +96,41 @@ func newTaskResponse(task *api.Task, containerMap map[string]*api.DockerContaine
 	}
 }
 
+func newContainerResponse(dockerContainer *api.DockerContainer, eni *api.ENI) v1.ContainerResponse {
+	container := dockerContainer.Container
+	resp := v1.ContainerResponse{
+		Name:       container.Name,
+		DockerId:   dockerContainer.DockerID,
+		DockerName: dockerContainer.DockerName,
+	}
+
+	for _, binding := range container.GetKnownPortBindings() {
+		port := v2.PortResponse{
+			ContainerPort: binding.ContainerPort,
+			Protocol:      binding.Protocol.String(),
+		}
+
+		if eni == nil {
+			port.HostPort = binding.HostPort
+		} else {
```

Comment:
If the container does use ENI, the `binding` should be empty, you should read the port information from `container.Port`. So I think you can do this:
```
bindings := container.GetKnownPortBindings()
if len(bindings) == 0{
// Put the port information from task definition
}else{
// put the information from bindings
}
```